### PR TITLE
Allow to set GitLab feature flags

### PR DESCRIPTION
**Pull request by tobiashuste**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -198,6 +198,24 @@ Choose whether GitLab's web-server Nginx redirects HTTP requests to HTTPS:
 nginx_redirect_http_to_https: 'false'
 ```
 
+#### Set GitLab feature flags
+
+Set [GitLab feature flags](https://docs.gitlab.com/ee/user/feature_flags.html)
+to enable or disable additional features.
+The variable is a list of key-value pairs which requires the `name` of the
+feature flag and its `status` that can be either `enable` or `disable`.
+The default value is se to an empty list `[]`.
```
Comment (christianhueserhzdr) [inline README.md]:
```suggestion
The default value is set to an empty list `[]`.
```

File: tasks/feature-flag.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  ansible.builtin.command:
+    cmd: 'gitlab-rails runner "is_feature_enabled = Feature.enabled?(:{{ gitlab_feature_flag.name }}); puts is_feature_enabled"'
+  register: "is_feature_enabled"
+  changed_when: false
+
+- name: "Enable or disable a feature flag"
+  ansible.builtin.command:
+    cmd: 'gitlab-rails runner "Feature.{{ gitlab_feature_flag.status }}(:{{ gitlab_feature_flag.name }})"'
+  when: "(is_feature_enabled.stdout == 'true' and gitlab_feature_flag.status == 'disable') or (is_feature_enabled.stdout == 'false' and gitlab_feature_flag.status == 'enable')"
+
```
Comment (christianhueserhzdr) [inline tasks/feature-flag.yml:18]:
```suggestion
```

File: tasks/feature-flag.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# SPDX-FileCopyrightText: 2023 Helmholtz Centre for Environmental Research (UFZ)
+# SPDX-FileCopyrightText: 2023 Helmholtz-Zentrum Dresden-Rossendorf (HZDR)
+#
+# SPDX-License-Identifier: Apache-2.0
+
+---
+
+- name: "Check if a feature flag is already enabled"
```
Comment (christianhueserhzdr) [inline tasks/feature-flag.yml:8]: I'd suggest to append the feature flag name to the task name to be able to identify which feature flag is handled:

```suggestion
- name: "Check if feature flag is already enabled for {{ gitlab_feature_flag.name }}"
```

File: tasks/feature-flag.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  ansible.builtin.command:
+    cmd: "gitlab-rails runner 'is_feature_enabled = Feature.enabled?(:{{ gitlab_feature_flag.name }}); puts is_feature_enabled'"
+  register: "is_feature_enabled"
+  changed_when: false
+
+- name: "Enable or disable a feature flag"
```
Comment (christianhueserhzdr) [inline tasks/feature-flag.yml:14]: I'd suggest to append the feature flag name to the task name to be able to identify which feature flag is handled:

```suggestion
- name: "Enable or disable feature flag {{ gitlab_feature_flag.name }}"
```

**Review by christianhueserhzdr — CHANGES_REQUESTED** [review]
Dear Tobias @tobiashuste, thanks for the nice implementation. I have only cosmetic suggestions. Do you think the Ansible output is easier to read if we know which of the feature flags is handled at a certain point in the pipeline? ansible-lint should be fine with appending variables at the end of task names.

**Review by christianhueserhzdr — APPROVED** [review]
Thanks, Tobias @tobiashuste, for the refinements. I will approve and merge this PR now.

File: tasks/main.yml [hunk 5]
```diff
@@ -35,4 +35,13 @@
   ansible.builtin.import_tasks: configure.yml
   become: yes
 
+- name: "Force all notified handlers to run at this point. Required for feature flags."
+  ansible.builtin.meta: "flush_handlers"
```
Comment (Normo) [inline tasks/main.yml:39]: We should be careful with flushing handlers. Did you check whether this has an impact on subsequent roles?

**Pull request merged**